src/main/java/com/blackducksoftware/integration/jira/task/setup/UpgradeSteps.java

```diff
@@ -72,8 +85,35 @@ public void upgradeToV6FromAny(final PluginSettings pluginSettings) {
         settingsWrapper.setProjectMappingsJson(config.getHubProjectMappingsJson());
     }
 
-    public static String getInstallDateString(final PluginSettings pluginSettings) {
-        return (String) pluginSettings.get(PluginConfigKeys.BLACKDUCK_CONFIG_JIRA_FIRST_SAVE_TIME);
+    // Delete when customers all upgrade to 4.2.0+
+    public void updateOldMappingsIfNeeded() {
+        final PluginConfigurationDetails pluginConfigDetails = new PluginConfigurationDetails(pluginSettings);
+        final BlackDuckJiraConfigSerializable config = new BlackDuckJiraConfigSerializable();
+        if (StringUtils.isBlank(pluginConfigDetails.getProjectMappingJson())) {
```

Comment:
We'll probably want to store the projectMappingJson to a string variable as we use it a couple times.